Pycloud Deployer/pycloud_deployer.py

```diff
@@ -0,0 +1,73 @@
+import os
+import argparse
+from dotenv import load_dotenv
+from azure.identity import DefaultAzureCredential
+from infra_classes import WebServer, DBServer, LoadBalancer
+
+# Load variables in .env file into ENVIRONMENT
+if os.environ.get('Environment') == 'development':
+    print("Loading environment variables from .env file")
+    load_dotenv(".env")
+
+# Obtain subscription ID
+subscription_id = os.environ.get('SUBSCRIPTION_ID')
+
+# Acquire a credential object
+token_credential = DefaultAzureCredential()
+
+parser = argparse.ArgumentParser(
+                    prog="Pycloud Deployer", 
+                    description="A Python CLI tool that facilitates the deployment of a 2 or 3 tier infrastructure to Azure.")
+
+# This lets us split up this program's functionality into sub-commands such as setup, info and teardown.
+# If a sub-command such as setup or info was entered, this sub-command would be stored and can be accessed in the attribute named 'command'
+subparsers = parser.add_subparsers(dest='command')
+
+# Setting up the 'setup' subcommand with its options
+setup_parser = subparsers.add_parser("setup", help="Deploys the infrastructure")
+setup_parser.add_argument("-d","--deployment-name", required=True, help="Name of the deployment")
+setup_parser.add_argument("-t","--tier", required=True, type=int, choices=[2,3], help="Specify the type of infrastructure")
+setup_parser.add_argument("--webservers", type=int, default=1, help="Specify the number of webservers to be deployed")
+
+# Setting up the 'info' subcommand with its options
+info_parser = subparsers.add_parser("info", help="Retrieve information about the infrastructure")
+info_parser.add_argument("-d","--deployment-name", help="Name of the deployment")
+
+# Setting up the 'teardown' subcommand with its options
+teardown_parser = subparsers.add_parser("teardown", help="Remove the entire deployed infrastructure")
+teardown_parser.add_argument("-d","--deployment-name", help="Name of the deployment")
```

Comment:
The argument -d/--deployment-name is repeated in each subparser. If this argument is common to all subcommands, it might be more efficient to define it once at the main parser level.